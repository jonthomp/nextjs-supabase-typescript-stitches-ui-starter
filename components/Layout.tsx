import Head from "next/head";
import { styled } from "../stitches.config";

const StyledLayout = styled("div", {
  minHeight: "100vh",
  padding: "0 $1",
  display: "flex",
  flexDirection: "column",
  justifyContent: "center",
  alignItems: "center",
});

const StyledMain = styled("div", {
  padding: "$6 0",
  flex: 1,
  display: "flex",
  flexDirection: "column",
  justifyContent: "center",
  alignItems: "center",
});

const StyledFooter = styled("div", {
  width: "100%",
  height: 100,
  borderWidth: "$1",
  borderTop: "solid #eaeaea",
  display: "flex",
  justifyContent: "center",
  alignItems: "center",
  "& img": {
    marginLeft: "$1",
  },
  "& a": {
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    textDecoration: "none",
  },
});

export const Layout = ({ children }) => {
  return (
    <StyledLayout>
      <StyledMain>{children}</StyledMain>

      <StyledFooter>
        <a
          href="https://vercel.com?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
          target="_blank"
          rel="noopener noreferrer"
        >
          Powered by{" "}
        </a>
      </StyledFooter>
    </StyledLayout>
  );
};
