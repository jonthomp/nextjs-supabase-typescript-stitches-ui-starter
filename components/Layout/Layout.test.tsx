import { render, screen, cleanup } from "@testing-library/react";
import { Layout } from "./Layout";

afterEach(cleanup);

describe("<Layout />", () => {
  it("Renders the default Layout with children", () => {
    const { asFragment } = render(<Layout>Hello world</Layout>);

    expect(screen.getByTestId("layout")).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });
});
