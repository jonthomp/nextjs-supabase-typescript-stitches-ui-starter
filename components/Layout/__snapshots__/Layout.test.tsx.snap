// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> Renders the default Layout with children 1`] = `
<DocumentFragment>
  <div
    class="c-hoGnYm"
    data-testid="layout"
  >
    <div
      class="c-kcbfpd"
    >
      Hello world
    </div>
    <div
      class="c-hdjAns"
    >
      <a
        href="https://vercel.com?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        Powered by 
      </a>
    </div>
  </div>
</DocumentFragment>
`;
